File: JavaScript/Easy/RankTransformOfAnArray.js
/**
 * @param {number[]} arr
 * @return {number[]}
 */
var arrayRankTransform = function (arr) {
  const ranks = new Map();
  const sorted = [...new Set(arr)].sort((a, b) => a - b);
  sorted.forEach((num, i) => ranks.set(num, i + 1));
  return arr.map((num) => ranks.get(num));
};

File: Python/Easy/RankTransformOfAnArray.py
from typing import List


class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        sorted_unique = sorted(set(arr))
        rank_map = {num: i + 1 for i, num in enumerate(sorted_unique)}
        return [rank_map[num] for num in arr]
